### PR TITLE
Add collapsed title padding for flexible space bar 69017

### LINKED ISSUE #69017: Different paddings in titlePading of FlexibleSpace
Hi!
image image

It is possible to have different padding in the titlePadding when it is collapsed and when it is not. 
When it is not collapsed I would like a 16px filling vertically, and when it is collapsed I would like a 50px filling because of the hamburger menu.

But as the paddings are the same for both situations, what happens in the images above happens.

My code is this:
```dart
SliverAppBar(
                    expandedHeight: 100.0,
                    floating: false,
                    pinned: true,
                    flexibleSpace: FlexibleSpaceBar(
                        centerTitle: false,
                        titlePadding:
                            EdgeInsets.symmetric(vertical: 16, horizontal: 16),
                        title: Text(
                            AppLocalizations.of(context).translate("dashboard"),
                            style: TextStyle(
                                color: Colors.white,
                                fontWeight: FontWeight.bold)),
                        background:
                            Container(color: Theme.of(context).primaryColor))),
```

I need help please. Thank you!

### PR DESCRIPTION
**Pull request by bobekos**
Added new possibility to animate a custom set title padding in expanded and collapsed mode when using inside the FlexibleSpaceBar.

Fixes #69017 

Before

https://github.com/user-attachments/assets/1730ec5a-e8f5-44bc-b981-8974fb5d4078

 After

https://github.com/user-attachments/assets/0d3f7e35-195e-41ae-bc79-939cdd03a062

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md
[test-exempt]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md
[Features we expect every widget to implement]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/blob/main/docs/contributing/Chat.md
[Data Driven Fixes]: https://github.com/flutter/flutter/blob/main/docs/contributing/Data-driven-Fixes.md

### CODE REVIEW

File: packages/flutter/lib/src/material/flexible_space_bar.dart [hunk 1]
```diff
@@ -137,6 +138,16 @@ class FlexibleSpaceBar extends StatefulWidget {
   /// padding of 72.0 pixels and bottom padding of 16.0 pixels.
   final EdgeInsetsGeometry? titlePadding;
 
+  /// Defines the padding applied to the [title]
+  /// when the [FlexibleSpaceBar] is collapsed.
+  ///
+  ///This allows for separate control of the title's padding
```
Comment (cedvdb) [inline packages/flutter/lib/src/material/flexible_space_bar.dart:144]: missing space after /// here and below

File: packages/flutter/lib/src/material/flexible_space_bar.dart [hunk 2]
```diff
@@ -205,6 +216,32 @@ class _FlexibleSpaceBarState extends State<FlexibleSpaceBar> {
     };
   }
 
+  EdgeInsetsGeometry _getTitlePadding(
+    double t,
```
Comment (cedvdb) [inline packages/flutter/lib/src/material/flexible_space_bar.dart:220]: "t" is not descriptive

File: packages/flutter/lib/src/material/flexible_space_bar.dart (hunk 1 above)
Comment (bobekos) [reply packages/flutter/lib/src/material/flexible_space_bar.dart:144]: Hi @cedvdb glad you found time to review the changes. What do you mean with missing space? Do you mean the blank lines on line 143 and 146? I simply follow the [code guide](https://github.com/flutter/flutter/blob/master/docs/contributing/Style-guide-for-Flutter-repo.md#leave-breadcrumbs-in-the-comments) and the description of other attributes like 'titlePadding'.

File: packages/flutter/lib/src/material/flexible_space_bar.dart (hunk 2 above)
Comment (bobekos) [reply packages/flutter/lib/src/material/flexible_space_bar.dart:220]: Fair enough this is true. Also here the is another method in the class "_getCollapsePadding" which also just passed the time attribute as 't'.  Do you have any suggestions how to call this?

File: packages/flutter/lib/src/material/flexible_space_bar.dart (hunk 1 above)
Comment (bobekos) [reply packages/flutter/lib/src/material/flexible_space_bar.dart:144]: Oh i am so sorry :D. I know what you mean. I will fix that.

File: packages/flutter/lib/src/material/flexible_space_bar.dart (hunk 2 above)
Comment (bobekos) [reply packages/flutter/lib/src/material/flexible_space_bar.dart]: Ok i change the progress 't' now to 'expansionFactor'. Let me know if you are ok with this?

File: packages/flutter/lib/src/material/flexible_space_bar.dart [hunk 3]
```diff
@@ -137,6 +138,16 @@ class FlexibleSpaceBar extends StatefulWidget {
   /// padding of 72.0 pixels and bottom padding of 16.0 pixels.
   final EdgeInsetsGeometry? titlePadding;
 
+  /// Defines the padding applied to the [title]
+  /// when the [FlexibleSpaceBar] is collapsed.
+  ///
+  /// This allows for separate control of the title's padding
+  /// when the [FlexibleSpaceBar] transitions from expanded to collapsed state.
+  ///
+  /// The [collapsedTitlePadding] is only used if [titlePadding] is set.
```
Comment (Piinks) [inline packages/flutter/lib/src/material/flexible_space_bar.dart:147]: This seems like odd behavior, why?

**Review by Piinks — COMMENTED** [review]
Looking at the video, I might expect this behavior to be handled automatically. Having the title overlap the menu button feels like a bug. Can FlexibleSpaceBar do this without the user needing to intervene?

**Comment by Piinks** [comment]
Yes, it looks like FlexibleSpaceBarSettings should have the information that there is a leading widget in the app bar so it will not overlap it.

**Comment by bobekos** [comment]
Hi @Piinks,

yes, this behaviour is already automatically adopted by FlexibleSpaceBarSettings. However, this only happens if no specific 'titlePadding' is set. With your own custom 'titlePadding', the system does not check whether there is a leading widget. 

```dart
//current 'master' implementation
final double leadingPadding = (settings.hasLeading ?? true) ? 72.0 : 0.0;
final EdgeInsetsGeometry padding = widget.titlePadding ?? EdgeInsetsDirectional.only(
     start: effectiveCenterTitle ? 0.0 : leadingPadding,
     bottom: 16.0,
);
```

Since I didn't want to change the existing behaviour, I thought that a new attribute would be better. The attribute 'collapsedTitlePadding' is only used if 'titlePadding' is set.

Of course, we could also automatically set the padding for a leading Widget if the custom 'titlePadding' is set. However, this would possibly break the current behaviour for some users as the leading padding is hardcoded to 72.0.

**Comment by Piinks** [comment]
> With your own custom 'titlePadding', the system does not check whether there is a leading widget.

We should see if we can handle this then.

> this would possibly break the current behaviour for some users

Maybe, we won't know unless we try it out and run through our tests to verify.

> I thought that a new attribute would be better

Adding new attributes increases complexity, and requires the user to handle more of the unique behaviors of this widget. Would you like to try so see if this can be handled automatically and see if it is in fact breaking? I don't imagine it is a desirable behavior currently, so it may be worth breaking to fix the bug. :)

**Comment by Piinks** [comment]
Hey @bobekos would you like to continue working on this PR?

**Comment by bobekos** [comment]
@Piinks 
Sorry for the late reply. Yes, I would like to finish it. I just don't have the time at the moment. I would try to finish it this week

**Comment by bobekos** [comment]
@Piinks 
I'm so sorry about that. I wasn't paying attention and put in all the changes since the Master as PR. I will delete the PR and reset it if necessary.

**Comment by victorsanni** [comment]
> I'm so sorry about that. I wasn't paying attention and put in all the changes since the Master as PR. I will delete the PR and reset it if necessary.

No worries, this happens all the time. i believe you can force push to a previous clean commit.

**Pull request closed without merging**